pkg/platform/model/checkpoints.go

```diff
@@ -78,15 +61,9 @@ func FetchLanguagesForCommit(commitID strfmt.UUID) ([]Language, error) {
 	for _, requirement := range checkpoint {
 		if NamespaceMatch(requirement.Namespace, NamespaceLanguageMatch) {
 			lang := Language{
-				Name:    requirement.Requirement,
-				Version: requirement.VersionConstraint,
-			}
-			if lang.Version == "" {
-				version, err := getVersionFromConstraints(requirement.VersionConstraints)
-				if err != nil {
-					return nil, errs.Wrap(err, "Unable to determine commit's language version based on constraints")
-				}
-				lang.Version = version
+				Name:               requirement.Requirement,
+				Version:            requirement.VersionConstraint,
+				versionConstraints: requirement.VersionConstraints,
```

Comment:
You can use `VersionConstraint` here. We should not be playing with the structured version constraints on the client as it makes us too opinionated and prone to backward incompatible changes.

I know it says `VersionConstraint` is deprecated; but this entire API will be replaced long before `VersionConstraint` will be fully dropped.

Reply:
I wish we could use `requirement.VersionConstraint` (that's what I tried originally), but if you `state init org/namespace --language python@3.10`, `VersionConstraint` will be 3.10 instead (which was requested) of 3.10.0. `VersionConstraints` contains 3.10.0 (which was resolved), so that's why the extra processing is needed.

Reply:
I'm not sure I follow. Both `VersionConstraint` and `VersionConstraints` will contain the requested version, not the resulting version. I guess we have some logic that translates it to a full semver perhaps? But either way it's still the requested version, which in this case just happens to be a full version. Point is this cannot be used to determine the resolved version, so it being incomplete shouldn't matter.